Desktop/utilities/languagemodel.cpp

```diff
@@ -336,7 +331,9 @@ void LanguageModel::findQmFiles(QString qmlocation)
 		{
 			Log::log() << "Language (" << loc.language() << ") not registered in LanguageModel, adding it now" << std::endl;
 
-			_languages.push_back(loc.language());
+			int i = 0;
+			while (i < _languages.size() && loc.nativeLanguageName().compare(_languagesInfo[_languages[i]].nativeLanguageName, Qt::CaseInsensitive) > 0)	i++;
```

Comment:
Why are you using a while loop to add random empty languages to the list? 

Why not make _languages a more sensible datatype...

Reply:
Or just throw it away and use the keys as they are in _languagesInfo?

Reply:
In any case, make the while loop into a for loop.

Reply:
I have restructured the LanguageModel class:

- There is no currentIndex anymore, only a languageName
- _languages map as for key the value displayed in the dropdown. This key is composed by the language code + the native name of the language. So the order of the language code will be used. And as the native language name is used, thanks to the language code, it is easier to know which language the name means (I have no clue to distinguish Japanese from Chinese).
- The locale country is also used: we can now differentiate a language spoken in different country (I know that the portuguese is now mainly Brazilan Portuguese, but in the future Portugal Portuguese could be added).
- Obsolete codes are removed.